### PR TITLE
Improve pre-training defaults for GRU

### PR DESCRIPTION
**Pull request by MorganCThomas**
- Updated the default config settings for more sensible options
- Properly implemented restricted SMILES randomization
- Can now load gzipped .smi files
- LR scheduler is now updated every step depending on step_size (not every epoch)

### CODE REVIEW

**Review by albertbou92 — COMMENTED** [review]
Looks good! Can we also adapt the `pretrain_distributed.py` script and run pre-commit?

**Comment by MorganCThomas** [comment]
Done? You mean just moving the lr scheduler?
